app/src/main/java/com/example/habitsmasher/UserValidator.java

```diff
@@ -99,10 +101,10 @@ public boolean isNewUserValid(String username, String email, String password) {
     }
 
     public boolean isLoginValid(String email, String password) {
-        return !(email.isEmpty() | password.isEmpty() |
-                !PatternsCompat.EMAIL_ADDRESS.matcher(email).matches() |
+        return !(password.isEmpty() |
                 password.length() < LOWER_CHARACTER_LIMIT |
-                password.length() > UPPER_CHARACTER_LIMIT);
+                password.length() > UPPER_CHARACTER_LIMIT |
+                !super.isEmailValid(email));
```

Comment:
Yup, this is basically what I meant we should be doing. But instead of it being a call to **super**, it can just be a call to an instance of `EmailValidator` that we keep locally inside the `UserValidator`